website/static/js/registrationUtils.js

```diff
@@ -471,6 +473,12 @@ RegistrationEditor.prototype.init = function(draft) {
     self.currentPage(self.draft().pages()[0]);
 
     self.serialized.subscribe(self.autoSave.bind(self));
+
+    $(window).on('beforeunload', function(e) {
+        if (self.isDirty()) {
```

Comment:
Note that because of how isDirty is defined, this will warn the user about unsaved changes even if they never entered text. Low priority but trivially annoying.